packages/storybook/src/components/DemoPage.tsx

```diff
@@ -0,0 +1,378 @@
+import {
+  convertOTelDocumentToSpanCards,
+  flattenSpans,
+} from "@ai-agent-trace-ui/data";
+import {
+  type OpenTelemetryDocument,
+  type TraceRecord,
+  type TraceSpan,
+} from "@ai-agent-trace-ui/types";
+import {
+  Button,
+  DetailsView,
+  SpanCardCollapseAllButton,
+  SpanCardExpandAllButton,
+  SpanCardSearchInput,
+  TraceList,
+  TraceListItemHeader,
+  TreeView,
+} from "@ai-agent-trace-ui/ui";
+import cn from "classnames";
+import { ArrowLeft } from "lucide-react";
+import { useCallback, useEffect, useMemo, useState } from "react";
+
+import quoTavAgentDataRaw from "../data/quo_tav_agent.json";
+import ragEarningsAgentDataRaw from "../data/rag_earnings_agent.json";
+import smolDeepResearchAgentDataRaw from "../data/smol_deep_research_agent.json";
+
+const quoTavAgentData = convertOTelDocumentToSpanCards(
+  quoTavAgentDataRaw as OpenTelemetryDocument[],
+);
+const ragEarningsAgentData = convertOTelDocumentToSpanCards(
+  ragEarningsAgentDataRaw as OpenTelemetryDocument[],
+);
+const smolDeepResearchAgentData = convertOTelDocumentToSpanCards(
+  smolDeepResearchAgentDataRaw as OpenTelemetryDocument[],
+);
+
+const MOCK_TRACE_RECORDS: TraceRecord[] = [
+  {
+    id: "quo-tav",
+    name: "7a8b9c1d",
+    spansCount: 24,
+    durationMs: 3200,
+    agentDescription: "research-agent",
+  },
+  {
+    id: "rag-earnings",
+    name: "f2e3d4c5",
+    spansCount: 156,
+    durationMs: 45670,
+    agentDescription: "data-analysis-bot",
+  },
+  {
+    id: "smol-deep-research",
+    name: "9b8a7c6d",
+    spansCount: 13,
+    durationMs: 2500,
+    agentDescription: "customer-support-ai",
+  },
+];
+
+// Recursive filtering function that preserves nested structure
+const filterSpansRecursively = (
+  spans: TraceSpan[],
+  searchValue: string,
+): TraceSpan[] => {
+  if (!searchValue.trim()) {
+    return spans;
+  }
+
+  return spans
+    .map((span) => {
+      // Check if current span matches
+      const currentSpanMatches = span.title
+        .toLowerCase()
+        .includes(searchValue.toLowerCase());
+
+      // Recursively filter children
+      const filteredChildren = span.children
+        ? filterSpansRecursively(span.children, searchValue)
+        : undefined;
+
+      // Check if any children match
+      const hasMatchingChildren =
+        filteredChildren && filteredChildren.length > 0;
+
+      // Keep span if it matches or has matching children
+      if (currentSpanMatches || hasMatchingChildren) {
+        return {
+          ...span,
+          children: filteredChildren,
+        };
+      }
+
+      // Filter out this span if neither it nor its children match
+      return null;
+    })
+    .filter((span): span is NonNullable<typeof span> => span !== null);
+};
+
+export const DemoPage = () => {
+  return (
+    <div className="h-full w-full p-4 lg:p-8">
+      <div className="hidden 2xl:!block">
+        <DesktopLayout />
+      </div>
+
+      <div className="2xl:hidden">
+        <MobileLayout />
```

Comment:
How difficult was it to create a single layout so that there would be no duplicate markup? How justified would this be in terms of code complexity?